schema.json

```diff
@@ -0,0 +1,125 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Thunderbird In-App Notifications",
+  "type": "array",
+  "items": {
+    "type": "object",
+    "properties": {
+      "id": {
+        "type": "string",
+        "format": "uuid",
+        "description": "Unique ID set by the server"
+      },
+      "start_at": {
+        "type": "string",
+        "format": "date-time",
+        "description": "Timestamp after which Thunderbird will show the event after startup."
+      },
+      "end_at": {
+        "type": "string",
+        "format": "date-time",
+        "description": "Timestamp after which Thunderbird will never show the event, even if it has never been shown to the user."
+      },
+      "title": {
+        "type": "string",
+        "description": "Short sentence describing the event which will be displayed in the Thunderbird UI."
+      },
+      "description": {
+        "type": "string",
+        "description": "A short paragraph that can contain HTML and will be displayed in the Thunderbird UI."
+      },
+      "URL": {
+        "type": "string",
+        "format": "uri",
+        "description": "URL to open from the CTA, if any."
+      },
+      "CTA": {
+        "type": "string",
+        "description": "Link text to show for the URL."
+      },
+      "severity": {
+        "enum": [1, 2, 3, 4, 5],
+        "description": "Severity level, where 1 is the most important/urgent and 5 is the least."
+      },
+      "type": {
+        "enum": ["donation", "message", "security", "blog"],
+        "description": "Category of notification."
+      },
+      "targeting": {
+        "type": "object",
+        "properties": {
+          "percent_chance": {
+            "type": "number",
+            "minimum": 0,
+            "maximum": 100
+          },
+          "exclude": {
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/profile"
+            }
+          },
+          "include": {
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/profile"
+            }
+          }
+        }
+      }
+    },
+    "required": [
+      "id",
+      "start_at",
+      "end_at",
+      "title",
+      "description",
+      "severity",
+      "type",
+      "targeting"
+    ],
+    "dependentRequired": { "CTA": ["URL"] }
+  },
+  "definitions": {
+    "profile": {
+      "type": "object",
+      "properties": {
+        "locales": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "versions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "channels": {
+          "type": "array",
+          "items": {
+            "enum": ["esr", "release", "beta", "daily"],
```

Comment:
I think this would be `default` for builds that aren't on any channel, like local builds or possibly other "unbranded" builds. The SoW doesn't have any mention of that, just trying to think of all the permutations of what we might encounter in the wild.

Reply:
That's a good point. @Sancus, any objection to sending notifications to any/all builds by specifying `default`?

Reply:
Not sure it was clear, I meant the string default, so `"default"`

Reply:
My intent was that everything in the targeting section be an optional filter. So you would simply not have a channel filter if you aren't targeting by channel, since it isn't relevant information.

Same for targeting all locales, versions, operating systems, whatever.

Reply:
:+1: 

Question: Should I add a "denylist" to the schema? (and maybe change "targeting" to "allowlist"?)

Reply:
Isn't that what `exclude` and `include` do?

Reply:
hahahaha yes.
:joy: